adds issue on access token: description, cause and solution (#42)
DaniMori/sortee-git-workshop: doc/Known_issues.md

---
title: "Known issues"
output: github_document
editor:
  markdown:
    wrap:      80

>     canonical: true

Reviewer: The default markdown configuration in the project [was changed to "sentence wrapping"](https://github.com/DaniMori/git-workshop/blob/87fd01c1130012a89f955d22fb9b64525f4f984d/rstudio_project.Rproj#L19), but this file was still using column wrapping instead.

I added the column wrapping configuration and reverted the "sentence wrapping" changes. Please **be mindful about what you add, and try not to bulk commit!** 😉

Unknown. Make sure that the {renv} environment is completely updated by running
`renv::status()`. Try to restore the environment running
`renv::restore(confirm = FALSE)` if the environment is not completely restored.

### Problem

> When the learner has had a GitHub account for long, and not used much, GitHub

Reviewer: For consistency with previous issues:

```suggestion
When the participant has had a GitHub account for long, and not used much, GitHub
```

Reviewer: Also:

```suggestion
When the learner has had a GitHub account for long, but have not used it too much lately, GitHub
```

may ask to use an "access token" to confirm their identity.

When trying to clone their own private repo then, instead of being prompted by

> Rstudio to authenticate with the browser, an authentication issues rises.

Reviewer: ```suggestion
Rstudio to authenticate with the browser, an authentication issue arises.
```

### Cause

> GitHub in July 2020, announced their intent to require the use of token-based

Reviewer: ```suggestion
In July 2020, GitHub announced their intention to require the use of token-based authentication [here](https://github.blog/news-insights/company-news/token-authentication-requirements-for-api-and-git-operations/)
```

Reviewer: Also, "intention"?

Reviewer: Can you also add the relevant link to "GitHub announced", please?

authentication for all authenticated Git operations. Beginning August 13, 2021,
they no longer accept account passwords when authenticating Git operations on
GitHub.com.

It affects the following workflows: - Command line Git access - Desktop
applications using Git (GitHub Desktop is unaffected) - Any apps/services that

> access Git repositories on GitHub.com directly using their password

Reviewer: Correct use of bullet points here

> ### Solution

Reviewer: I have a concern with this solution.

Although I find it perfectly valid, I understand that it happens because they may have an old password authentication token cached in their credential management system (in Windows or other system?). IMO, to make it easier and more consistent, it would be preferable, if possible, to try to "authenticate with the browser" as we usually do with the rest of the users in the workshop. In order to do that, they should "clean the credentials" from the credential management system.

I think I have done this before, but I'm not 100% sure I would be able to reproduce the steps. This solution should be easier though, and IMO worth adding as a first step to try. What do you think? I could explore it further.

Author: Hmmm... It's just that maybe it happens without any other passwords being stored. The first time it happened to me, I had nothing stored and no token created, but that was because they had built in two-step verification (and your password isn't enough). Great if you can explore it further; I can't see how to make it more replicable than what I wrote at the time.

Reviewer: Not sure this solution is 100% correct, but I think it should go more ore less like this:

```suggestion
### Solution

**Alternative 1:**

1. Open "Windows Credential Manager": Go to the Startup menu, type "Windows Credential Manager" and hit ENTER.
2. In section "Generic credentials" look for a credential for the domain "https://github.com" (or similar).
3. Click on the right downward arrow to open the credential details.
4. Click on "Remove".
5. In the confirmation dialogue, click "Yes".
6. Try to follow again the steps to clone the repository from Rstudio.
```

It's possible that we cannot test it until it happens "in the wild" 😕

Author: Then? Can we wait until this situation happens, or should we commit this?

Reviewer: Not sure yet. What do you think? Can we include it, maybe provisionally, and open an issue to "check this solution when it happens" (or leave the issue open for now)?

Author: Maybe its best to leave the issue open for now

Reviewer: Ok, commit this change then. I updated [the description](#issue-3203381155) to avoid closing [the issue](../issues/41) automatically (and I think we're done with this PR! 🥳 )

> Learners should use a personal access token

Reviewer: ```suggestion
Participants should use a personal access token
```

Alternatively, change previous references to "learners"?

Reviewer: Check out [the second warning here](https://docs.github.com/en/get-started/git-basics/caching-your-github-credentials-in-git#git-credential-manager)

(<https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens>)

> The easiest solution is to create a perosnal access token (classic):

Reviewer: Suggestion: Collapse lines 106-109 into one (they are redundant), and link the text "create a personal access token" instead of adding the link in parentheses afterwards.

-   Verify your email address, if it hasn't been verified yet.
-   In the upper-right corner of any page on GitHub, click your profile photo,
    then click `Settings`.
-   In the left sidebar, click Developer settings.
-   In the left sidebar, under Personal access tokens, click Tokens (classic).
-   Select Generate new token, then click Generate new token (classic).
-   In the "Note" field, give your token a descriptive name.
-   To give your token an expiration, select Expiration, then choose a default
    option or click Custom to enter a date.
-   Select all the scopes.
-   Click Generate token.

> -   Optionally, to copy the new token to your clipboard, click .

Reviewer: Suggestion: Numbered list instead of bullet points?

Author: With bullet points is perfectly ok!

Reviewer: When the actions to follow are a "sequence", "perfectly ok" for me means to provide a notion that they should be followed "sequentially", and that's what a numbered list is for 😜 
But I said "suggestion" and bullet points is acceptable for me, so up to you 😉

Unknown. Make sure that the {renv} environment is completely updated by running
`renv::status()`. Try to restore the environment running
`renv::restore(confirm = FALSE)` if the environment is not completely restored.

### Problem

When the participant has had a GitHub account for long, and not used much, GitHub
may ask to use an "access token" to confirm their identity.

When trying to clone their own private repo then, instead of being prompted by

> Rstudio to authenticate with the browser, an authentication issue arises.

Reviewer: Seeing the previous "problems", I see they are expressed as simple "one-liners" stating the problem very concisely. To keep that style consistent, I'm thinking now that I'd suggest rewriting this as (more or less):

```suggestion
When trying to clone their own private repo, GitHub asks the user for an "access token" to confirm their identity.
```
(Did I understand it right?)

And then put the rest in the section "Cause". Does it sound reasonable?

### Cause

In July 2020, GitHub announced their intention to require the use of token-based authentication [here](https://github.blog/news-insights/company-news/token-authentication-requirements-for-api-and-git-operations/)

> authentication for all authenticated Git operations. Beginning August 13, 2021,

Reviewer: ```suggestion
for all authenticated Git operations. Beginning August 13, 2021,
```
